src/giskardpy/tree_manager.py

```diff
@@ -0,0 +1,140 @@
+                self.node.insert_child(manager_node.node, idx)
+                self.enabled_children.add(manager_node)
+
+        def remove_child(self, manager_node):
+            if isinstance(self.node, PluginBehavior):
+                if manager_node in self.enabled_children:
+                    self.enabled_children.remove(manager_node)
+                    self.node.remove_plugin(manager_node.node.name)
+                elif manager_node in self.disabled_children:
+                    self.disabled_children.remove(manager_node)
+                else:
+                    raise RuntimeError('could not remove node from parent. this probably means that the tree is inconsistent')
+            else:
+                if manager_node in self.enabled_children:
+                    self.enabled_children.remove(manager_node)
+                    self.node.remove_child(manager_node.node)
+                elif(manager_node in self.disabled_children):
+                    self.disabled_children.remove(manager_node)
+                else:
+                    raise RuntimeError('could not remove node. this probably means that the tree is inconsistent')
+                idx = self.disabled_children.bisect_right(manager_node)
+                for c in list(self.disabled_children.islice(start=idx)):
+                    c.position -= 1
+                idx = self.enabled_children.bisect_right(manager_node)
+                for c in list(self.enabled_children.islice(start=idx)):
+                    c.position -= 1
+
+    def __init__(self, tree):
+        self.tree = tree
+        self.tree_nodes = {}
+        self.init_map(tree.root, None, 0)
+
+    def init_map(self, node, parent, idx):
+        manager_node = TreeManager.ManagerNode(node=node, parent=parent, position=idx)
+        if parent != None:
+            parent.enabled_children.add(manager_node)
+        if isinstance(node, PluginBehavior):
+            children = node.get_plugins()
+            for child_name in children.keys():
+                child_node = TreeManager.ManagerNode(node=children[child_name], parent=manager_node, position=0)
+                self.tree_nodes[child_name] = child_node
+                manager_node.enabled_children.add(child_node)
+        self.tree_nodes[node.name] = manager_node
+        for idx, child in enumerate(node.children):
+            self.init_map(child, manager_node, idx)
+
+
+    def disable_node(self, node_name):
+        t = self.tree_nodes[node_name]
+        if t.parent != None:
+            return t.parent.disable_child(t)
+        else:
+            logging.logwarn('cannot disable root node')
+            return False
+
+
+    def enable_node(self, node_name):
+        t = self.tree_nodes[node_name]
+        if t.parent != None:
+            t.parent.enable_child(t)
+        else:
+            logging.loginfo('root node')
+
+
+    def insert_node(self, node, parent_name, position=-1):
+        if node.name in self.tree_nodes.keys():
```

Comment:
`if node.name in self.tree_nodes:` is faster because `.keys()` creates a list first.